whistl/datasets.py

```diff
@@ -442,16 +431,84 @@ def __len__(self):
         return len(self.labels)
 
 
+def create_sample_to_platform_mapping(metadata):
+    '''Generate a dictionary mapping each sample to the platform that generated it
+
+    Arguments
+    ---------
+    metadata: json
+        A json object containing the metadata for a study
+
+    Returns
+    -------
+    sample_to_platform: dict
+        A dictionary mapping each sample accession to the platform used to sequence the sample
+    '''
+    sample_to_platform = {}
+
+    experiment_metadata = metadata['experiments']
+    sample_metadata = metadata['samples']
+
+    sample_list = []
+    for experiment in experiment_metadata:
+        try:
+            experiment_samples = experiment_metadata[experiment]['sample_accession_codes']
+            sample_list.extend(experiment_samples)
+        except KeyError as e:
+            # If an experiment doesn't have any samples for some reason, skip it
+            print(e)
+            pass
+
+
+    for sample in sample_list:
```

Comment:
That's a good idea. In this case it would either push me past my line limit, or make me change my indentation style, but I will look for the pattern in the future